core/src/main/scala/org/apache/spark/sql/sources/StoreStrategy.scala

```diff
@@ -52,9 +52,9 @@ object StoreStrategy extends Strategy {
     case DMLExternalTable(name, storeRelation: LogicalRelation, insertCommand) =>
       ExecutedCommand(ExternalTableDMLCmd(storeRelation, insertCommand)) :: Nil
 
-    case InsertIntoTable(l @ LogicalRelation(t: RowPutRelation, _),
-    part, query, overwrite, false) if part.isEmpty =>
-      ExecutedCommand(PutIntoDataSource(l, query, overwrite)) :: Nil
+    case InsertIntoTable(l@LogicalRelation(t: RowPutRelation, _),
```

Comment:
I guess this still does not fix the PUT conversion of INSERT statements , right ?